### DISCUSSION TITLE
Custom CSS is not overriding the module css

### DISCUSSION THREAD

**Discussion by manishoctal** [vercel/next.js#51884] [Help]
### Summary

As per the screenshot, I have a layout and inside I am importing the `bootsrap.css` after that I just importing the `globals.css` and `responsive.css`
But CSS is not overriding the bootstrap.css, It seems, `globals.css` and `responsive.css` are not loaded, I have checked the consoles as well. but not loading. 
My NextJS version is: 13.4.5

package.json

```
{
  "name": "website",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "axios": "^1.4.0",
    "bootstrap": "^5.3.0",
    "classnames": "^2.3.2",
    "dayjs": "^1.11.8",
    "eslint": "8.42.0",
    "eslint-config-next": "13.4.5",
    "jwt-decode": "^3.1.2",
    "lodash": "^4.17.21",
    "next": "13.4.5",
    "react": "18.2.0",
    "react-bootstrap": "^2.7.4",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.45.0",
    "react-hot-toast": "^2.4.1",
    "react-icons": "^4.10.1",
    "react-tag-input-component": "^2.0.2"
  }
}

```
 
![image](https://github.com/vercel/next.js/assets/43563293/03167756-fc66-4c4c-9b01-7bc853ebfd95)

**Comment by icyJoseph** [comment]
And you have verified this inside the auth section of the app, right?

**Reply by manishoctal** [reply]
Right, actually generated layout.css is loading bootstrap after my custom css

**Reply by icyJoseph** [reply]
Alright, so they load, but, out of order?

**Reply by manishoctal** [reply]
Yes, exactly

**Reply by icyJoseph** [reply]
Could you try to upgrade to the latest next? Let's make sure it hasn't been fixed already

**Reply by manishoctal** [reply]
I tried, but it is same situation here.

**Comment by icyJoseph** [comment]
I'll give it a go soon. I'm worried this might be a problem with route groups mmm

**Reply by manishoctal** [reply]
I am stuck in the middle of the project and my project deadline are meeting soo. Is there any hack on this or any solution?

**Reply by icyJoseph** [reply]
Mmm I tried just now, without route groups, and I see no issue.  My global CSS file is being placed after bootstraps.

Gonna add route groups now.

**Reply by icyJoseph** [reply]
And yeah, it works with route groups too. 

https://github.com/vercel/next.js/assets/21013447/5f859e13-00b6-4020-9897-59694ffc411f

And you can see, the CSS I've defined comes last:

![Screenshot 2023-06-27 at 20 47 28](https://github.com/vercel/next.js/assets/21013447/7ea1b6f1-e09b-4715-9e43-80b2a03888a0)

**Reply by manishoctal** [reply]
Check my video, still globals.css loading before bootstrap.css 
What can be an issue? 
https://github.com/vercel/next.js/assets/43563293/9198b31a-675d-4b5c-961f-a92c437d26c4

**Reply by icyJoseph** [reply]
Weird, I noticed that when I restart the server, it is always out of order. And I need to reload the page to get it in order. And then it works, even in dev mode. 

I am really not sure what is going on with your CSS though.  

If you are running out of time to debug this, try with:

```css
@import "~bootstrap/dist/css/bootstrap.css";
```

At top of the global css file.

**Reply by manishoctal** [reply]
I tried both possible ways, but unlucky it is not working, even I am trying by deleting .next folder all the time.

![image](https://github.com/vercel/next.js/assets/43563293/45f19ed2-6532-433e-8c74-a70fcfe87a4f)
![image](https://github.com/vercel/next.js/assets/43563293/35b6f37d-fe92-485f-adef-c49bc558e338)

**Reply by icyJoseph** [reply]
Mmm I wonder if you are somehow importing that bootstrap package elsewhere? and maybe that causes a bad CSS order resolution?

**Reply by manishoctal** [reply]
I am not importing elsewhere only in layout.js

**Reply by manishoctal** [reply]
@icyJoseph Any resolution possible at this moment? I am wondering, If I deploy the same app to vercel, then order is seems correct. My Custom CSS overriding the bootstrap.

**Reply by icyJoseph** [reply]
Aha, wow that's even weirder. 

Do you use Windows? Vercel deploys use Linux so maybe that's the reason?

**Reply by manishoctal** [reply]
yes, right now I am using windows for development.

**Reply by icyJoseph** [reply]
Mmm I really don't think it should matter, but if the deploy is working fine 🤷‍♂️

**Reply by manishoctal** [reply]
But  it is very hard to see bad css layout during development causing bad mindset. :'(

**Reply by icyJoseph** [reply]
Could you try to start a new project, install Bootstrap and just in one page try this? Same behavior?

**Comment by BrunoFerreira95** [comment]
With tailwind, it's aready working in started project with CLI, the people don't like of tailwind but turme faster.
